### ISSUE TITLE
AssertionError: Argument `in_features`=50 is not divisble be `phm_dim`4

### ISSUE THREAD

**Issue by wtzhao1631** [bayer-science-for-a-better-life/phc-gnn#8]
Hi，

When I run the code via bash run_script_hiv_phm4.sh

I get the following error：

Traceback (most recent call last):
  File "train_hiv.py", line 636, in <module>
    main()
  File "train_hiv.py", line 569, in main
    model = UPH_SC_ADD(phm_dim=phm_dim, learn_phm=learn_phm, phm_rule=phm_rule,
  File "/ogb/phc-gnn-master/phc/hypercomplex/undirectional/models.py", line 134, in __init__
    self.convs[i] = PHMMessagePassing(in_features=in_dim, out_features=out_dim, bias=bias,
  File "/ogb/phc-gnn-master/phc/hypercomplex/undirectional/messagepassing.py", line 494, in __init__
    self.transform = PHMGINEConvSoftmax(in_features, out_features, phm_dim, phm_rule, learn_phm,
  File "/ogb/phc-gnn-master/phc/hypercomplex/undirectional/messagepassing.py", line 277, in __init__
    self.transform = PHMMLP(in_features=in_features, out_features=out_features, phm_dim=phm_dim,
  File "/phc-gnn-master/phc/hypercomplex/layers.py", line 324, in __init__
    self.linear1 = PHMLinear(in_features=in_features, out_features=int(factor*out_features),
  File "/ogb/phc-gnn-master/phc/hypercomplex/layers.py", line 231, in __init__
    assert in_features % phm_dim == 0, f"Argument `in_features`={in_features} is not divisble be `phm_dim`{phm_dim}"
AssertionError: Argument `in_features`=50 is not divisble be `phm_dim`4

The default embedding dimension is 200. Does this cause this error?

**Comment by tuanle618** [comment]
Dear @wtzhao1631 ,
thanks for your interest in our work and running the experiments.
The issue was related to the PR #6 I made to make the computation of the PHMLinear layer faster. 
In summary: a few GNN modules were affected by it, which I did not update properly. I did a few changes in a recent PR #10 , which I have merged with the main branch.
I tested the benchmark bash script. It should work now.

Let me know, if you have further questions. 

Best regards,
Tuan
